build-new-version.sh

```diff
@@ -52,7 +55,11 @@ download() {
 dlfile() {
     component="$1"
     triple="$2"
-    download "https://static.rust-lang.org/dist/$component-$TARGET_VERSION-$triple.tar.gz"
+    if [ x"$TARGET_VERSION" == x"nightly" -a "$NIGHTLY_DATE" ]; then
```

Comment:
I'd suggest extracting this test out to a helper function like `is_nightly`.

Reply:
OK, I'll do that.